singularity-eos/eos/eos.hpp

```diff
@@ -129,7 +129,19 @@ class Gruneisen : public EosBase<Gruneisen> {
             const Real b, const Real rho0, const Real T0, const Real P0, const Real Cv,
             const Real rho_max)
       : _C0(C0), _s1(s1), _s2(s2), _s3(s3), _G0(G0), _b(b), _rho0(rho0), _T0(T0), _P0(P0),
-        _Cv(Cv), _rho_max(rho_max) {}
+        _Cv(Cv), _rho_max(rho_max) {
+    // Warn user when provided rho_max is greater than the computed rho_max
+#ifndef NDEBUG
+    const Real computed_rho_max = ComputeRhoMax(s1, s2, s3, rho0);
+    if (rho_max > RHOMAX_SAFETY * computed_rho_max) {
+      std::cout << "WARNING: Provided rho_max, " << rho_max
```

Comment:
This means that it can't be a `__device__` constructor -- need to use printf.

Reply:
I don't think we should assume construction on host, with the possible exception of tablular EOSs.  In case we want to do that, we should not label the constructor as a `PORTABLE_INLINE_FUNCTION`.  It could very much make sense to just call the constructor on the device for some applications.

Reply:
I agree but I'm not sure everything is marked appropriately. That may be something to look at later.